### PR TITLE
TECH: Address some Recall and RecallController JPA related warnings

### PR DESCRIPTION
**Pull request by AndyNewton**
TECH: Address some Recall and RecallController JPA related warnings plus remove unused code

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt [hunk 1]
```diff
@@ -378,11 +377,7 @@ class Api {
   )
 }
 
-data class Pdf(val content: String) {
-  companion object {
-    fun encode(content: ByteArray): Pdf = Pdf(content.encodeToBase64String())
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RecallController.kt:383]: Idea said unused and build passes with removed.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt [hunk 2]
```diff
@@ -93,8 +93,10 @@ data class Recall(
   val vulnerabilityDiversityDetail: String? = null,
   @Enumerated(STRING)
   val mappaLevel: MappaLevel? = null,
+  @Embedded
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt:96]: Fixes JPA related warning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt [hunk 3]
```diff
@@ -307,6 +309,7 @@ data class SentencingInfo(
   }
 }
 
+@Suppress("JpaAttributeMemberSignatureInspection") // Suppressing erroneous Idea error: https://youtrack.jetbrains.com/issue/IDEA-240844
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Recall.kt:312]: See Idea error link from code comment.

**Review by philsegal — APPROVED** [review]
Not so keen on Embedded, but 🤷‍♂️

**Pull request merged**